### ISSUE TITLE
Incorrect 'The variable 'X' is assigned but its value is never used' when constant is used in attribute of local function

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#60645]
I don't think the warning makes sense here:

```csharp
public class C
{
    public int P
    {
        get
        {
            // warning CS0219: The variable 'X' is assigned but its value is never used
            const int X = 5;
            return local1();

            [My(X)]
            int local1() => 0;
        }
    }
}

public class MyAttribute : System.Attribute
{
    public MyAttribute(int i) { }
}
```

Lambda attributes have the same bug too:

```csharp
public class C
{
    public int P
    {
        get
        {
            // warning CS0219: The variable 'X' is assigned but its value is never used
            const int X = 5;
            var f = [My(X)] () => 0;
            return f();
        }
    }
}

public class MyAttribute : System.Attribute
{
    public MyAttribute(int i) { }
}
```

**Comment by Dweeberly** [comment]
FWIW this also shows up here:
```
const string X = "X";
static string F(string x = X) { return x; }
```
X is reported as never used (CS0219).

**Comment by Youssef1313** [comment]
I'm looking into this.
